packages/components/src/DateInput/DateInput/DateInput.tsx

```diff
@@ -27,6 +27,7 @@ export const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(
     <div className={classnames(styles.dateInput, classNameOverride)}>
       <Label
         htmlFor={id}
+        id={`${id}-input-label`}
```

Comment:
How hard would it be to avoid the loose coupling here and instead init this string once and use it in the two places?

Reply:
It's slightly awkward but I thought it was a better alternative than introducing and drilling a new prop down through the DateInputWithIconButon then into the DateInput since `id` is already in use

Reply:
To you're point though it is something we can easily update in the `DateRangePicker` since it sits all the same level